core/src/Realm/UserRealmDiscoveryResponse.cs

```diff
@@ -0,0 +1,67 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using Microsoft.Identity.Core.Helpers;
+using Microsoft.Identity.Core.Http;
+
+namespace Microsoft.Identity.Core.Realm
+{
+    [DataContract]
+    internal sealed class UserRealmDiscoveryResponse
+    {
+        [DataMember(Name = "ver")]
+        public string Version { get; set; }
+
+        [DataMember(Name = "account_type")]
+        public string AccountType { get; set; }
+
+        [DataMember(Name = "federation_protocol")]
+        public string FederationProtocol { get; set; }
+
+        [DataMember(Name = "federation_metadata_url")]
+        public string FederationMetadataUrl { get; set; }
+
+        [DataMember(Name = "federation_active_auth_url")]
+        public string FederationActiveAuthUrl { get; set; }
+
+        [DataMember(Name = "cloud_audience_urn")]
+        public string CloudAudienceUrn { get; set; }
+
+        internal static async Task<UserRealmDiscoveryResponse> CreateByDiscoveryAsync(string userRealmUri, string userName, RequestContext requestContext)
+        {
+            var msg = "Sending request to userrealm endpoint.";
```

Comment:
>"Sending request to userrealm endpoint." [](start = 22, length = 40)

at some point in time we agreed to move all texts used to a single file.... @travis likely recall where that ended

Reply:
I asked Travis the other day about this. He said we have not started this effort yet, because we encountered some issues on managing a resource file.
If you really have a strong desire to fix this one right now, I would rather suggest we completely remove this message because: (1) it is for logging only; (2) it is not very useful anyway; we already know every WIA will always trigger a UserRealmDiscovery. I'll mark it as pending for now.

---
In reply to: [201544441](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1118#discussion_r201544441) [](ancestors = 201544441)